### ISSUE TITLE
Update function app source zip file

### ISSUE THREAD

**Issue by ashisa** [Azure/commercial-marketplace-data-offers#11]
The current version is a test version. Update the zip file after we have finalized the function app code.

**Comment by code4clouds** [comment]
@ashisa - Would you mind adding more context to this issue?

**Comment by ashisa** [comment]
The source of the notification endpoint function app is located at commercial-marketplace-data-offers/powershell-data-function. The publisher ARM template deployment deploys the code using a zip of the function app source code that is located at /install/publisher-azure/functionapp.

Ideally, every time we make changes to the code at /powershell-data-function, we need to update the zip file at /install/publisher-azure folder. However, since we are modifying the code to accommodate more features etc, we need to be cognizant that before have a publisher deploy the template, we ensure the zip file is updated with the latest code changes.

**Comment by code4clouds** [comment]
Got it!  Thanks for the clarification.

We had a similar situation in another project that pushed a zip with source to Azure Websites, all in the same repo.

This is how we resolved it:

1. Use Powershell to create the zip use the latest version.
2. Have the deployment script create the zip file at runtime with the latest code from commercial-marketplace-data-offers/powershell-data-function /power-data-function 
3. Copy the newly created zip file to /install/publisher-azure/functionapp
4. Push the zip to the Azure resource


You can see an example here (line 68):
https://github.com/Azure/Microsoft-commercial-marketplace-transactable-SaaS-offer-SDK/blob/master/deployment/Templates/Deploy.ps1


Do not create the zip file and store it in GH (frown upon in OSS) as its not source.

**Comment by dstarr** [comment]
Makes sense. And I like the technique. The code to zip is already there.

I try to only have images for docs as far as binaries go.
